src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java

```diff
@@ -501,13 +528,46 @@ private ProcurementEvent getCurrentEvent(final ProcurementProject project) {
 
   private void addProjectUserMapping(final String jaggaerUserId, final ProcurementProject project,
       final String principal) {
-    var projectUserMapping = ProjectUserMapping.builder().project(project).userId(jaggaerUserId)
-        .timestamps(createTimestamps(principal)).build();
-    retryableTendersDBDelegate.save(projectUserMapping);
+
+    var existingUserMapping = retryableTendersDBDelegate
```

Comment:
This block could be simplified with use of `Optional.ifPresentOrElse(...)` I think..

```
retryableTendersDBDelegate.findProjectUserMappingByProjectIdAndUserId(project.getId(), jaggaerUserId).ifPresentOrElse(userMapping -> {
// Update existing mapping and save
},() -> {
// Create new user mapping and save
}

);
```